client/core/core.go

```diff
@@ -7446,3 +7488,218 @@ func (c *Core) findActiveOrder(oid order.OrderID) (*trackedTrade, error) {
 	}
 	return nil, fmt.Errorf("could not find active order with order id: %s", oid)
 }
+
+// fetchFiatExchangeRates starts the fiat rate fetcher goroutine and schedules
+// refresh cycles. Use under ratesMtx lock.
+func (core *Core) fetchFiatExchangeRates() {
+	if core.stopFiatRateFetching != nil {
+		core.log.Debug("Fiat exchange rate fetching is already enabled")
+		return
+	}
+	ctx, cancel := context.WithCancel(core.ctx)
+	core.stopFiatRateFetching = cancel
+
+	core.wg.Add(1)
+	go func() {
+		defer core.wg.Done()
+		tick := time.NewTimer(0) // starts rate fetching immediately.
+		for {
+			select {
+			case <-ctx.Done():
+				tick.Stop()
+				return
+			case <-tick.C:
+				core.refreshFiatRates(ctx)
+			}
+			tick = core.nextRateSourceTick()
+		}
+	}()
+}
+
+// refreshFiatRates refreshes the fiat rates for rate sources whose values have
+// not been updated since fiatRateRequestInterval.
+func (core *Core) refreshFiatRates(ctx context.Context) {
+	ctx, cancel := context.WithTimeout(ctx, 4*time.Second)
+	defer cancel()
+
+	var wg sync.WaitGroup
+	core.ratesMtx.Lock()
+	for _, source := range core.fiatRateSources {
+		if time.Since(source.lastTry()) > fiatRateRequestInterval {
+			wg.Add(1)
+			go func(source *commonRateSource) {
+				defer wg.Done()
+				source.refreshRates(ctx, core.assetMap())
+			}(source)
+		}
+	}
+	core.ratesMtx.Unlock()
+	wg.Wait()
+
+	fiatRatesMap := core.fiatConversions()
+	if len(fiatRatesMap) != 0 {
+		core.notify(newFiatRatesUpdate(fiatRatesMap))
+	}
+}
+
+// nextRateSourceTick checks the fiat rate source's last request, and calculates
+// when the next cycle should run.
+func (core *Core) nextRateSourceTick() *time.Timer {
+	core.ratesMtx.Lock()
+	defer core.ratesMtx.Unlock()
```

Comment:
`core.fiatRateSources` will be used _far_ more frequently for read-only purposes, so we'd benefit from a `sync.RWMutex` instead.